From caf92fe51445de125eb3d78f17f6e539bcf3d4bf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Wed, 26 Jul 2017 13:12:37 +0200
Subject: [PATCH] gir: Update glib annotations

One of the annotation changes that was pulled in for 1.53.4 broke
the introspection ABI and was therefore reverted in master. Pick
up the revert now so downstreams can more easily pull in the fix.
---
 gir/glib-2.0.c | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/gir/glib-2.0.c b/gir/glib-2.0.c
index 21bb4957..fe3c262e 100644
--- a/gir/glib-2.0.c
+++ b/gir/glib-2.0.c
@@ -4183,8 +4183,7 @@
  *
  * Defines the log domain.
  *
- * For applications, this is typically left as the default %NULL
- * (or "") domain. Libraries should define this so that any messages
+ * Libraries should define this so that any messages
  * which they log can be differentiated from messages from other
  * libraries and application code. But be careful not to define
  * it in any public header files.
@@ -4193,6 +4192,10 @@
  * |[
  * AM_CPPFLAGS = -DG_LOG_DOMAIN=\"Gtk\"
  * ]|
+ *
+ * Applications can choose to leave it as the default %NULL (or "")
+ * domain. However, defining the domain offers the same advantages as
+ * above.
  */
 
 
@@ -18154,7 +18157,7 @@
  * The array of returned groups will be %NULL-terminated, so
  * @length may optionally be %NULL.
  *
- * Returns: (array zero-terminated=1 length=length) (transfer full): a newly-allocated %NULL-terminated array of strings.
+ * Returns: (array zero-terminated=1) (transfer full): a newly-allocated %NULL-terminated array of strings.
  *   Use g_strfreev() to free it.
  * Since: 2.6
  */
@@ -18237,7 +18240,7 @@
  * be found, %NULL is returned and @error is set to
  * #G_KEY_FILE_ERROR_GROUP_NOT_FOUND.
  *
- * Returns: (array zero-terminated=1 length=length) (transfer full): a newly-allocated %NULL-terminated array of strings.
+ * Returns: (array zero-terminated=1) (transfer full): a newly-allocated %NULL-terminated array of strings.
  *     Use g_strfreev() to free it.
  * Since: 2.6
  */
@@ -18851,7 +18854,7 @@
  * Note that this function never reports an error,
  * so it is safe to pass %NULL as @error.
  *
- * Returns: (array length=length): a newly allocated string holding
+ * Returns: a newly allocated string holding
  *   the contents of the #GKeyFile
  * Since: 2.6
  */
-- 
2.13.3

